### PR TITLE
[PIP]Implement T-45 Power Armor

### PR DESCRIPTION
**Pull request by justinjohnson14**

### CODE REVIEW

File: Mage.Sets/src/mage/cards/t/T45PowerArmor.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    protected T45PowerArmorEffect(mage.cards.t.T45PowerArmorEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        Permanent creature = game.getPermanent(source.getSourcePermanentIfItStillExists(game).getAttachedTo());
+        game.getPermanent(source.getSourcePermanentIfItStillExists(game).getAttachedTo()).untap(game);
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/t/T45PowerArmor.java:84]: need to null check. just remove this line, and after the existing null check, `creature.untap(game)`

File: Mage.Sets/src/mage/cards/t/T45PowerArmor.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            creature.addCounters(CounterType.findByName(
+                    chosen.toLowerCase(Locale.ENGLISH)
+            ).createInstance(), source.getControllerId(), source, game);
+        }
+
+        return true;
+    }
+
+    @Override
+    public mage.cards.t.T45PowerArmorEffect copy() {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/t/T45PowerArmor.java:104]: no need to specify mage.cards.t here. I recommend using the gen-card.pl script to take care of the boilerplate

File: Mage.Sets/src/mage/cards/t/T45PowerArmor.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+}
+
+class T45PowerArmorEffect extends OneShotEffect {
+
+    private static final Set<String> choices = new LinkedHashSet<>(Arrays.asList("Menace", "Trample", "Lifelink"));
+
+    T45PowerArmorEffect() {
+        super(Outcome.BoostCreature);
+        staticText = "untap eqipped creature, then put your choice of a menace, trample, or lifelink counter on it";
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/t/T45PowerArmor.java:73]: typo - equipped

**Review by xenohedron — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/t/T45PowerArmor.java (hunk 2 above)
Comment (justinjohnson14) [reply Mage.Sets/src/mage/cards/t/T45PowerArmor.java:104]: I did use gen-card.pl not sure if it did something weird or if it got changed inadvertently while editing.

Comment (xenohedron) [reply Mage.Sets/src/mage/cards/t/T45PowerArmor.java]: ah good. yeah, this is a separate class not in scope of gen-card.pl, apologies for misleading. probably happened when it was copy-pasted from some other code as a template, just gotta double check everything you copy paste.

**Review by xenohedron — APPROVED** [review]

**Pull request merged**